resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/internal/SemaphoreBulkhead.java

```diff
@@ -94,7 +95,11 @@ public boolean isCallPermitted() {
 
     @Override
     public void onComplete() {
-        semaphore.release();
+        try {
```

Comment:
Can event publishing be optional ? We are currently not consuming any bulkhead events, and have no immediate plans to, so this will immediately become just a gc pressure for us ...

Reply:
@gibffe  Actually, it is optional. Check out `publishBulkheadEvent` method it won't publish anything if there is no subscriptions.

Reply:
@storozhukBM agree